dams-client/src/api.rs

```diff
@@ -70,47 +74,74 @@ impl Password {
     }
 }
 
-/// A `Session` is an abstraction over a
-/// communication session between an asset owner and a key server
-/// that provides mutual authentication, confidentiality, and integrity.
-/// An open `Session` is required to interact with this API.
-///
-/// A session can be ended manually, or it might time out and require
-/// re-authentication (that is, creation of a new [`Session`]).
+/// A `DamsClient` is an abstraction over client operations. It is used
+/// to hold state, common infrastructure, and a `tonic` client object.
+/// Authenticating to the DAMS returns a `DamsClient`, and this object
+/// must be passed to the remaining API functions.
 ///
 /// TODO #30: This abstraction needs a lot of design attention.
 #[derive(Debug)]
 #[allow(unused)]
-pub struct Session {
+pub struct DamsClient {
     session_key: [u8; 64],
+    config: Config,
+    tonic_client: DamsRpcClient<Channel>,
 }
 
-impl Default for Session {
-    fn default() -> Self {
-        Session {
-            session_key: [0; 64],
-        }
-    }
+/// Options for actions the client can take.
+pub(crate) enum ClientAction {
+    Register,
+    Authenticate,
 }
 
 #[allow(unused)]
-impl Session {
+impl DamsClient {
+    /// Connect to the gRPC client and return it to the client app.
+    ///
+    /// The returned client should be stored as part of the [`DamsClient`]
+    /// state.
+    async fn connect(address: String) -> Result<DamsRpcClient<Channel>, SessionError> {
+        if address.starts_with("https:") {
```

Comment:
```suggestion
    async fn connect(address: Uri) -> Result<DamsRpcClient<Channel>, SessionError> {
        if address.scheme() == Some(&Scheme::HTTPS) {
```
I think this should take a [Uri](https://docs.rs/http/latest/http/uri/struct.Uri.html) instead of a String. It's a bit more explicit and `tonic` can take it in `DamsRpcClient::connect`. 

`http::uri::Uri` is re-exported as `tonic::transport::Uri` so it might be good to use the `tonic` import in case the versions diverge.

It might also be convenient to store it as a `Uri` in the config, but `Uri` doesn't support `serde` so we'd have to work around that. I know a really nice trick for this if you want to hop on a call and go over it. We could also just add a `server_uri()` function to the config that calls `Uri::from_str(self.server_location)`.